src/useIsShake.ts

```diff
@@ -10,6 +11,8 @@ export function useIsShake() {
   const [subscription, setSubscription] = useState<Subscription | null>(null);
   const [isShakeTriggered, setIsShakeTriggered] = useState(false);
   const [isShakeReady, setIsShakeReady] = useState(false);
+  const [isAccelerometerAvail, setIsAccelerometerAvail] = useState<boolean | null>(null);
+  const [isAccelerometerAvailPending, setIsAccelerometerAvailPending] = useState(false);
```

Comment:
State for storing status of pending operations to check for or request permission of accelerometer availability. Prevents functions from being called multiple times if other `useEffect` dependencies are updated while `isAccelerometerAvail` is in the process of update.